### ISSUE TITLE
Overly-long gene names cause panaroo-extract-gene to fail

### ISSUE THREAD

**Issue by cednotsed** [gtonkinhill/panaroo#271]
I'm running panaroo on a set of prokka gff files, which include custom gene annotations (--proteins flag). 
I then extracted the gene sequences using the panaroo-extract-gene utility in panaroo, but encountered an error indicating the output file name is too long: 
```
Traceback (most recent call last):
  File "/nemo/lab/matthewsp/home/shared/cedric/utils/miniconda3/envs/prokka/bin/panaroo-extract-gene", line 10, in <module>
    sys.exit(main())
  File "/nemo/lab/matthewsp/home/shared/cedric/utils/miniconda3/envs/prokka/lib/python3.7/site-packages/panaroo/extract_gene_fasta.py", line 130, in main
    idtype=args.idtype,
  File "/nemo/lab/matthewsp/home/shared/cedric/utils/miniconda3/envs/prokka/lib/python3.7/site-packages/panaroo/extract_gene_fasta.py", line 80, in generate_fasta
    with open(outputfile, "w") as outfile:
OSError: [Errno 36] File name too long: '/nemo/lab/matthewsp/home/shared/cedric/ARG_host_jumps/results/pangenome_out/panaroo_out/cluster_108/core_genes/MEG_526cluster_2~~~MEG_606cluster_108~~~MEG_575cluster_108~~~MEG_604cluster_108~~~MEG_639cluster_108~~~MEG_541cluster_108~~~MEG_7917cluster_108~~~MEG_605cluster_108~~~MEG_7918cluster_108~~~MEG_526cluster_1~~~MEG_623cluster_108~~~MEG_7965cluster_108~~~MEG_517cluster_108~~~MEG_624cluster_108~~~MEG_507cluster_108~~~MEG_520cluster_108~~~MEG_532cluster_108~~~MEG_533cluster_108~~~MEG_628cluster_108~~~MEG_625cluster_108~~~MEG_516cluster_108~~~MEG_518cluster_108~~~MEG_522cluster_108~~~MEG_653cluster_108~~~MEG_627cluster_108~~~MEG_529cluster_108~~~MEG_7956cluster_108~~~MEG_636cluster_108~~~MEG_7950cluster_108~~~MEG_524cluster_108~~~MEG_7969cluster_108~~~MEG_609cluster_108~~~MEG_538cluster_108~~~MEG_527cluster_108~~~MEG_7916cluster_108~~~MEG_641cluster_108~~~MEG_528cluster_108~~~MEG_629cluster_108~~~MEG_573cluster_108~~~MEG_7925cluster_108~~~MEG_667cluster_108~~~MEG_669cluster_108~~~MEG_7964cluster_108~~~MEG_554cluster_108~~~MEG_535cluster_108~~~MEG_601cluster_108~~~MEG_537cluster_1~~~MEG_537cluster_2~~~MEG_537cluster_108~~~MEG_7909cluster_108~~~MEG_598cluster_108~~~MEG_574cluster_108~~~MEG_549cluster_108~~~MEG_512cluster_108~~~MEG_658cluster_108~~~MEG_565cluster_108~~~MEG_7928cluster_108.fasta'
```

This error arises because the multiple distinct gene annotations have been collapsed into the same gene family by panaroo, causing the output file name of the gene fasta to exceed OS-specific limits to the length of file names. 

I've overcome this issue by directly modifying `extract_gene_fasta.py` so that it truncates file names that exceed 50 characters:

```
         for line in infile:
             line = line.strip().split(",")
             if line[0] in args.queries:
                 geneids = set()
                 for genome, genes in zip(genomes, line[3:]):
                     genes = genes.replace("_len", "").replace("_pseudo", "").split(";")
                     for g in genes:
                         geneids.add((genome,g))

                 ## MY EDITS TO DEAL WITH LONG FILE NAMES ##
                 s = line[0]
                 s = (s[:50] + '~') if len(s) > 50 else s
                 line[0] = s
                 ##

                 generate_fasta(
                     geneids,
                     outputfile=args.output_dir + line[0] + ".fasta",
                     genedata=args.gene_data,
                     isdna=args.isdna,
                     idtype=args.idtype,
                 )
```

**Comment by gtonkinhill** [comment]
Thanks for reporting this issue. We'll try and add a fix soon.

@nzmacalasdair are you able to take a look at this? I'm still sorting out my server access

**Comment by nzmacalasdair** [comment]
Looks similar to the issue during gene alignment, I'll try and have a look this afternoon.

**Comment by nzmacalasdair** [comment]
This should be fixed in [0f0cd6f](https://github.com/gtonkinhill/panaroo/commit/0f0cd6f3fbc44b96c556389a5ad66b5c707aca18). Thanks for letting us know, @cednotsed!
